language/bytecode_verifier/src/type_memory_safety.rs

```diff
@@ -127,6 +131,43 @@ impl<'a> TypeAndMemorySafetyAnalysis<'a> {
         existing_borrows.is_empty()
     }
 
+    fn type_formals(&self) -> &[Kind] {
```

Comment:
Maybe a small comment about this since it seems to be a point of discussion. 

Something like
```
// Gives the current constraints on the type formals in the current scope/function 
```